sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -388,7 +389,7 @@ public static partial class DelegatedTokenCredential
     }
     public partial class DiagnosticsOptions
     {
-        internal DiagnosticsOptions() { }
+        protected internal DiagnosticsOptions() { }
```

Comment:
Why does this need to become protected?

Reply:
so that sub-classes can [construct an instance](https://github.com/Azure/azure-sdk-for-net/pull/27520/files#diff-b3cb569ad3e8d590b1044115386277b72b0144f5aeecd34f2bc189c40f373e60R20)